config_parser_test.cc

```diff
@@ -1,11 +1,71 @@
+#include <sstream>
+#include <string>
 #include "gtest/gtest.h"
 #include "config_parser.h"
 
-TEST(NginxConfigParserTest, SimpleConfig) {
-  NginxConfigParser parser;
-  NginxConfig out_config;
 
-  bool success = parser.Parse("example_config", &out_config);
+// Test fixture for testing parsing a string literal
+class NginxParseStringConfigTest : public ::testing::Test {
+protected:
+	bool parse_string(const std::string config_string) {
+		std::stringstream config_stream(config_string); 
+		return parser_.Parse(&config_stream, &config_);
+	}
+	NginxConfigParser parser_;
+	NginxConfig config_;
+};
 
-  EXPECT_TRUE(success);
+TEST_F(NginxParseStringConfigTest, ValidSimpleStatement) {
+	EXPECT_TRUE(parse_string("listen 80;"));
 }
+
+TEST_F(NginxParseStringConfigTest, MissingSemicolonSimpleStatement) {
+	EXPECT_FALSE(parse_string("worker_processes 4"));
+}
+
+TEST_F(NginxParseStringConfigTest, CurlyBraceBlock) {
+	EXPECT_TRUE(parse_string("server { listen 8080; }"));
+}
+
+TEST_F(NginxParseStringConfigTest, NestedCurlyBraceBlocks) {
+	const std::string config_string = \
+		"server { \n"
+			"\tlocation / { \n"
+				"\t\tproxy_pass http://localhost:8080; \n"
+			"\t}\n"
+		"}";
+
+	EXPECT_TRUE(parse_string(config_string))
+		<< config_string;
+}
+
+TEST_F(NginxParseStringConfigTest, EmptyCurlyBlock) {
```

Comment:
nginx does allow empty location {} blocks, and unless we have a case-by-case situation or the instructors say something different I'm going to follow what nginx goes with. Plus it gives us a test case to enforce whatever the decision is in the future.